pyarn/lexer_wrapper.py

```diff
@@ -37,6 +43,12 @@ def token(self):
         if t is None:
             return t
 
+        if t.type == 'STRING' and not t.lexer.is_quoted:
+            if t.value == 'false':
+                return Boolean(False)
+            if t.value == 'true':
+                return Boolean(True)
```

Comment:
I think there is another way to handle ambiguous tokens, which is to define t_BOOLEAN (as a function!) before t_STRING in the lexer file: https://www.dabeaz.com/ply/ply.html#ply_nn6
> All tokens defined by functions are added in the same order as they appear in the lexer file.

Not 100% sure it would work, but if it does, I would prefer that over this wrapper

Reply:
If it would be defined before t_STRING, we would parse half of `trueThat` as a boolean, and then `That` would be a string since we are not relying on the boundaries of the strings in the lexer. Then we would need to let the lexer look one character ahead to check for a `\n` or EOF and we would need to change the INDENT/DEDENT logic.

Alternatively, we can move the logic into the `t_STRING` function, checking for the string value and changing the type where needed. Thoughts?

Reply:
Would checking for word boundary in the regex solve this? `r'(true|false)\b'`

Reply:
Then the token would include whatever `\b` is, which we may use to determine indentation

Reply:
We could also try cloning the parser (https://www.dabeaz.com/ply/ply.html#ply_nn6) to look ahead. But I think handling it in the `t_STRING` function will look cleaner, let me push something so we can discuss on top of that.

Reply:
Isn't `\b` 0-length? https://docs.python.org/3/library/re.html

Reply:
d'oh!

Changed. I added an additional `\b` in the beginning just to be safe. However, if we'd get to the point during tokenizing that it would actually be needed, something would already be quite wrong there

Reply:
Actually, maybe handling it in t_STRING is better after all :D

I realized `\b` matches any non-word character, so `true@v1` would still get half-parsed as BOOLEAN even though it is a STRING. You could still do it with a lookahead assertion (`r"(true|false)(?=\s|$)"`), but at that point, your solution is probably cleaner

Reply:
Good catch :) done